pkg/types/clusterconfig/errors.go

```diff
@@ -204,6 +205,13 @@ func ErrorDidNotMatchStrictS3Regex() error {
 	})
 }
 
+func ErrorNATRequiredWithPrivateSubnetVisibility() error {
+	return errors.WithStack(&errors.Error{
+		Kind:    ErrNATRequiredWithPrivateSubnetVisibility,
+		Message: fmt.Sprintf("a nat gateway is required with private subnets; either set %s to %s or %s, or set %s to %s", NATGatewayKey, s.UserStr(SingleNAT), s.UserStr(HighlyAvailableNAT), SubnetVisibilityKey, s.UserStr(PublicSubnetVisibility)),
```

Comment:
I think something along these lines may make it easier to understand `a nat gateway is required with private subnets` -> ```a nat gateway is required when `subnet_visibility: private` is specified```